File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
def solution(numbers):
    global answer
    answer = 0
    visited = [0] * len(numbers)
    case = set()
    
    def prime(n):
        
        if n == 1 or n == 0:
            return False
        
        for i in range(2, int(n **(1/2))+1):
            if n % i == 0:
                return False
        return True
    
    def dfs(sm):

        if sm:
            case.add(int(sm))
        
        for i in range(len(numbers)):
            if visited[i] == 0:
                visited[i] = 1
                dfs(sm+numbers[i])
                visited[i] = 0
                
    dfs('')
    
    for i in case:
        if prime(i):
            answer += 1
    
    return answer

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function solution(numbers) {
    var answer = 0;
    const N = numbers.length
    let cases = new Set()
    let visited = new Array(N).fill(0) 
    
    function prime(x) {
        
        if (x === 1 || x === 0) {
            return false
        }
        
        for (let i = 2; i <= x ** (1/2); i++) {
            if (x % i === 0) {
                return false
            }
        }
        return true
    }
    
    
    function dfs(sm) {
        
        if (sm != '') {
            cases.add(parseInt(sm))
        }
        
        for (let i = 0; i < N; i++ ) {
            if (visited[i] == 0) {
                visited[i] = 1
                dfs(sm+numbers[i])
                visited[i] = 0
            }
        }    
    }
    
    dfs('')
    
    cases.forEach(x => {
        if (prime(x)) {
            answer++
        }
    })
    
    return answer;
}